src/platform/ESP32/ESP32FactoryDataProvider.cpp

```diff
@@ -198,12 +198,13 @@ CHIP_ERROR ESP32FactoryDataProvider::GetHardwareVersionString(char * buf, size_t
 CHIP_ERROR ESP32FactoryDataProvider::GetRotatingDeviceIdUniqueId(MutableByteSpan & uniqueIdSpan)
 {
     ChipError err = CHIP_ERROR_WRONG_KEY_TYPE;
-#if CHIP_ENABLE_ROTATING_DEVICE_ID && defined(CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID)
```

Comment:
Where was this option set previously `CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID`, if someone enabled rotating device id support

Reply:
It was not a config option, it was a byte array, please see below: https://github.com/project-chip/connectedhomeip/blob/a8d12af4be0cbf26f3f1352a5562eb7856ca8651/src/include/platform/CHIPDeviceConfig.h#L283-L293

Since we are reading it from the factory partition we won't be checking if that macro is defined or not.